### PR TITLE
DAOS-15497 client: set rpath in libioil.so and libpil4dfs.so

### PR DESCRIPTION
**Pull request by wiliamhuang**
In daos build test, scons sets env "LD_LIBRARY_PATH" including build directory. With setting rpath in libioil.so and libpil4dfs.so, dependent libraries will be loaded from system lib path instead of build directory.

Features: dfuse

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by wiliamhuang** [comment]
daos build with libioil.so using "--jobs 12" passed.
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-14042/1/artifact/Functional%20on%20EL%208.8/dfuse/daos_build.py/job.log
Command export D_IL_MAX_EQ=0;export PATH=/tmp/daos_dfuse_test_dfuse_daos_build_wt_il_1/venv/bin:$PATH;export VIRTUAL_ENV=/tmp/daos_dfuse_test_dfuse_daos_build_wt_il_1/venv;export COVFILE=/tmp/test.cov;export LD_PRELOAD=/usr/lib64/libioil.so;export D_LOG_FILE=/var/tmp/daos_testing/daos-il.log;export DD_MASK=all;export DD_SUBSYS=all;export D_LOG_MASK=WARN,IL=WARN;scons -C /tmp/daos_dfuse_test_dfuse_daos_build_wt_il_1/daos --jobs 12 completed in 7:29

File: utils/rpms/daos.spec [hunk 1]
```diff
@@ -351,6 +352,10 @@ mv test.cov{,-build}
       %{?scons_args}                  \
       %{?compiler_args}
 
+# setting rpath for libioil.so and libpil4dfs.so is needed for parallel compiling in daos build test
+patchelf --force-rpath --set-rpath %{_libdir} %{buildroot}/%{_libdir}/libpil4dfs.so
+patchelf --force-rpath --set-rpath %{_libdir} %{buildroot}/%{_libdir}/libioil.so
+
```
Comment (ashleypittman) [inline utils/rpms/daos.spec:358]: This is a build step so should go in scons code rather than the rpm spec file.

**Review by ashleypittman — CHANGES_REQUESTED** [review]
This is a great find, this has long been a problem that we haven't understood with the additional benefit that if we can fix this then it'll also reduce test times.

> In daos build test, scons sets env "LD_LIBRARY_PATH" including build directory.

I wouldn't have expected this to be the case, a good start would be to look into why this is, and if it's required for the entire build or just parts of it.  https://github.com/daos-stack/daos/blob/master/site_scons/site_tools/daos_builder.py#L67-L70 is a possible candidate.

We need to get to the bottom of this problem properly which is why I'm giving this a -1 review for now but having identified a potential fix then this PR is already a huge step forwards.

**Comment by wiliamhuang** [comment]
Closing this PR since Ashley has a better solution which fixes the issue in scons build script to eliminate unneeded setting "LD_LIBRARY_PATH", https://github.com/daos-stack/daos/pull/14049

**Pull request closed without merging**